File: backend/src/main/java/com/cosec/backend/models/Comment.java
package com.cosec.backend.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "comments")
public class Comment {

    @Id
    private String id;
    private String userId;
    private String username;
    private String caffId;
    private String text;

    public Comment(String id, String userId, String username, String caffId, String text) {
        this.id = id;
        this.userId = userId;
        this.username = username;
        this.caffId = caffId;
        this.text = text;
    }

    public Comment(String userId, String username, String caffId, String text) {
        this.userId = userId;
        this.username = username;
        this.caffId = caffId;
        this.text = text;
    }

    public Comment() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getCaffId() {
        return caffId;
    }

    public void setCaffId(String caffId) {
        this.caffId = caffId;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
}

File: caff_shop_app/lib/app/models/comment.dart
class Comment {
  final String id;
  final String userId;
  final String caffId;
  final String text;

  Comment({
    required this.id,
    required this.userId,
    required this.caffId,
    required this.text,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json["id"],
      userId: json["userId"],
      caffId: json["caffId"],
      text: json["text"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "userId": userId,
      "caffId": caffId,
      "text": text,
    };
  }
}
